cdap-docs/developers-manual/source/pipelines/developing-plugins/creating-a-plugin.rst

```diff
@@ -466,6 +482,13 @@ In order to implement a Streaming Source Plugin, you extend the ``StreamingSourc
   stream or dataset is not a macro.
 - ``getStream()``: Returns the ``JavaDStream`` that will be used as a source in the pipeline.
 
+.. rubric:: Lineage
+
+The lineage is registered using the ``referenceName`` provided when invoking ``registerLineage()`` on
+``StreamingContext`` in ``getStream()``. Note that the ``referenceName`` should be a valid ``DatasetId``.
+Optionally, you can extend from ``ReferenceStreamingSource``, which verifies the validity
```

Comment:
We cannot extend from the ReferenceStreamingSource since this in spark-plugin jar which will bring back all the plugins if included as a dependency. We should have a consistent explanation for all the Sources and Sinks. If we intend them to make it available via a jar then all the plugin types should work in the same manner. Otherwise we should avoid the mention of reference and include the right way to write a plugin and helps track lineage.